doc/api.md

```diff
@@ -1,128 +1,109 @@
 
-<!-- TOC WIP -->
-
 # Preface
 
-The **IoT Agent node library** offers a simple REST API which provides common functionality to access, provision and
-decommission devices.
+The IoT Agent mission is to provide a common abstraction layer between the devices and the NGSI entities stored in
+Context Broker. In order to achieve this, the IoT Agent sits between the Context Broker and a set of devices. It is in
+charge of translating the information coming from the devices into NGSI requests and viceversa.
 
-<!-- TOPICS WIP -->
+The **IoT Agent node library** is a Node.js module that can be used to implement IoT Agents. It provides a set of common
+functionality that can be used to implement the different IoT Agents, offering a simple REST API which provides common
+functionality to access, provision and decommission devices and groups of devices. This document describes the API
+provided by the IoT Agent node library.
 
 # Topics
 
 ## Terminology
 
-### Service groups
-
-### Devices
-
-### Provision
-
-## Device attributes model
-
-### `attributes`
-
-### `static_attributes`
-
-### `commands`
-
-### `lazy` attributes
-
-## Service Group API
-
-For some services, there will be no need to provision individual devices, but it will make more sense to provision
-different service groups, each of one mapped to a different type of entity in the context broker. How the type of entity
-is assigned to a device will depend on the Southbound technology (e.g.: path, port, APIKey...). Once the device has an
-assigned type, its configuration values can be extracted from those of the type.
-
-The IoT Agents provide two means to define those service groups:
-
--   Static **Type Configuration**: configuring the `ngsi.types` attribute within the `config.js` file.
--   Dynamic **Configuration API**: making use of the API URLs in the configuration URI, `/iot/services`. Please, note
-    that the configuration API manage servers under a URL that requires the `server.name` parameter to be set (the name
-    of the IoT Agent we are using). If no name is configured `default` is taken as the default one.
+-   **Devices**: A resource that match physical devices that are connected to the IoT Agent. Each device has a set of
+    attributes that can be read or written and a set of commands that can be invoked. The device is identified by a
+    `device_id` and points particular entity in the context broker.
+-   **Config Groups**: Also known as `provisioning groups` or `service groups`. A logical group of devices. Each Config
+    Group has a set of attributes that can be read or written. The config group is identified by a an `apikey`, used to
+    authenticate the requests coming from the devices.
+-   **Measurements**: A set of values that are sent by a device to the IoT Agent.
+-   **Service**: It is the `FIWARE-Service` that the device or config group belongs to.
+-   **Subservice**: It is the specific `FIWARE-ServicePath` that the device or config group belongs to.
```

Comment:
```suggestion
-   **Service**: It is the `fiware-service` that the device or config group belongs to.
-   **Subservice**: It is the specific `fiware-servicepath` that the device or config group belongs to.
```

(As far as I remember we always use the headers in lowercase in the requests API section, although I could be wrong)

Reply:
[Orion API document](https://github.com/telefonicaid/fiware-orion/blob/master/doc/manuals/orion-api.md) uses `Fiware-Service` and `Fiware-ServicePath`

Reply:
I have edited my suggestion above accordingly.

Reply:
...so it matches the original document :) Never mind.

NTC